subsys/lorawan/lorawan.c

```diff
@@ -595,7 +595,7 @@ int lorawan_send(uint8_t port, uint8_t *data, uint8_t len,
 	int ret = 0;
 	bool empty_frame = false;
 
-	if (data == NULL) {
+	if (len && data == NULL) {
```

Comment:
I'd suggest to make the `len` check more explicit and swap the conditions, so that the second condition doesn't have to be evaluated in most cases:

```suggestion
	if (data == NULL && len > 0) {
```

Reply:
Thanks for the suggestion, I have just updated my PR.